### ISSUE TITLE
Need template/process for communicating accessibility related aspects of design

### ISSUE THREAD

**Issue by seanforyou23** [patternfly/patternfly-design#828]
## Design template needed for identifying accessible names in interactive components

I'd like to propose we introduce a somewhat formal means of identifying accessibility-related design concepts in design deliverables to be used as part of the design documentation for every component. Ultimately, this would be an enhancement to the existing design process.

Today, it's almost always the case that we end up with duplicate accessible names for critical interactive controls, which makes the accessible experience (AX) not great. For example, we end up with views that are represented as like below (Notice the duplicate text "Actions collapsed popup buton" which doesn't distinguish between table items)

<img width="1680" alt="actions-kebab-accessible-name" src="https://user-images.githubusercontent.com/5942899/76550278-41b55280-6468-11ea-88d2-8d71b194dbdf.png">

This problem exists across *many* of our existing components and it would be good to solve them all in a simple and same manner.

Here's an example of a simple way we can document what the accessible names should be for component elements that may have many instances on a same page.

![accessible-name-design-documentation](https://user-images.githubusercontent.com/5942899/76550665-dfa91d00-6468-11ea-97af-2c9a3fe68e9d.png)

The main thing to notice in this example is that the accessible name for the edit icon button uses a **format** which is the text "Edit" followed by the name of the thing it is going to edit. This is critical because, without it, every edit button simply reads "Edit" to assistive technology - which doesn't signify which thing the user is actually editing.

This format should be used for all labels in similar interactions like expanding/deleting/selecting/etc.

IMO the definition for done for this issue would be to have delivered said design documentation for the existing components which currently exhibit this problem. These cases are easily identifiable by driving the documentation site with a screen reader of your choice.

What do others think? Is it something we can commit to or want to do at all? Ping @mcarrano & @mmenestr

**Comment by jgiardino** [comment]
👍 

An example of one method where we are handling a similar situation (e.g. many actions of the same kind for different elements) is with Remove button in the [ChipGroup](https://www.patternfly.org/v4/documentation/react/components/chipgroup) component. In that case, we use `aria-labelledby` to create an accessible name for the Remove button. 

It would be helpful to capture what techniques are possible (`aria-label` vs `aria-labelledby`) for meeting this guideline. 

Also, regarding the syntax to use for labels, this WCAG item provides some guidance (i.e. the visible label of the element must be stated first, followed by any additional text): https://www.w3.org/WAI/WCAG21/quickref/#label-in-name

**Comment by mcarrano** [comment]
Thanks for filing this issue @seanforyou23 . Sounds like this is an issue of educating consumers of PatternFly about best practices and what's available to help make pages more accessible. Perhaps the first step is to create more/better documentation. Not sure if this belongs as a design guideline, as part of the component documentation or both. Do you think there is also an opportunity to create a Katacoda tutorial about this?

Because this mostly involves documentation IMO, I am going to move this issue over to patternfly-org for further tracking. If you think there are changes we need to make to any components to achieve this, we can open separate issues.

cc @LHinson @rachael-phillips
